pkg/storage/s3/s3.go

```diff
@@ -372,15 +452,13 @@ func (d *driver) CompleteConfiguration(cr *opapi.ImageRegistry, modified *bool)
 		return err
 	}
 
-	if len(d.Config.Bucket) == 0 {
-		d.Config.Bucket = cfg.Storage.S3.Bucket
-	}
-
 	if len(d.Config.Region) == 0 {
 		d.Config.Region = cfg.Storage.S3.Region
 	}
 
-	cr.Status.Storage.State.S3 = d.Config
-	*modified = true
+	cr.Status.Storage.S3 = d.Config
+
+	d.CreateStorage(cr, modified)
```

Comment:
why are you invoking create storage here?  I think the pattern we want is:

1) completeconfiguration is used to populate a boostrapped CR
2) then the operator sees the bootstrapped CR (or user provided CR) and makes it happen

we shouldn't tie "bootstrapping the CR" together with actually creating the storage, i don't think.